internal/controller/connector_controller.go

```diff
@@ -0,0 +1,338 @@
+			for _, configMapName := range configMapNames {
+				if err := appsv1beta1.DeleteAssociatedConfigMap(ctx, r.Client, types.NamespacedName{
+					Namespace: connector.Namespace,
+					Name:      configMapName,
+				}); err != nil {
+					log.Error(err, "fail to delete ConfigMap which generated by Connector",
+						"Connector", connector.Name,
+						"ConfigMap", configMapName,
+					)
+
+					return ctrl.Result{}, err
+				}
+			}
+
+			// clear generated Deployments if Connector is being deleted.
+			if err := r.deleteAssociatedDeployments(ctx, connector); err != nil {
+				log.Error(err, "fail to delete Deployment which generated by Connector",
+					"Connector", connector.Name,
+				)
+
+				return ctrl.Result{}, err
+			}
+
+			controllerutil.RemoveFinalizer(&connector, connectorfinalizerName)
+			if err := r.Update(ctx, &connector); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ConnectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&appsv1beta1.Connector{}).
+		Complete(r)
+}
+
+func (r *ConnectorReconciler) mergePatchesIntoConfigs(ctx context.Context, connector appsv1beta1.Connector) ([]map[string]interface{}, error) {
+	templateConfigMapName := appsv1beta1.GenConnectorConfigMapName(connector.Spec.TemplateName, true)
+	templateConfigMapNamespacedName := types.NamespacedName{
+		Namespace: connector.Namespace,
+		Name:      templateConfigMapName,
+	}
+
+	var templateConfigMap corev1.ConfigMap
+	if err := r.Get(ctx, templateConfigMapNamespacedName, &templateConfigMap); err != nil {
+		log.Error(err, "fail to fetch ConfigMap which stores ConnectorTemplate config")
+
+		return nil, err
+	}
+
+	var templateConfig map[string]interface{}
+	err := json.Unmarshal([]byte(templateConfigMap.Data["config"]), &templateConfig)
+	if err != nil {
+		log.Error(err, "fail to unmarshal ConnectorTemplate config")
+
+		return nil, err
+	}
+
+	var configs []map[string]interface{}
+
+	for _, stream := range connector.Spec.Streams {
+		config := make(map[string]interface{})
+		for k, v := range templateConfig {
+			config[k] = v
+		}
+
+		config["stream"] = stream
+
+		var patches map[string]map[string]interface{}
+		if connector.Spec.Patches != nil {
+			json.Unmarshal(connector.Spec.Patches, &patches)
+		}
+
+		if connector.Spec.Patches != nil {
+			if val, ok := patches[stream]; ok {
+				for k, v := range val {
+					config[k] = v
+				}
+			}
+		}
+
+		configs = append(configs, config)
+	}
+
+	return configs, nil
+}
+
+func (r *ConnectorReconciler) createConnectorDeployment(ctx context.Context, connector appsv1beta1.Connector, stream, configMapName string) error {
+	name := appsv1beta1.GenConnectorDeploymentName(connector.Name, stream)
+
+	return r.Create(ctx, &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: connector.Namespace,
+			Name:      name,
+		},
```

Comment:
I think should add ownerReference and some labels

Reply:
[c1e4d16](https://github.com/hstreamdb/hstream-operator/pull/59/commits/c1e4d162cd13df954be6ca0076d1251814c48261)

Reply:
I didn't see ownerReference and some labels for this deployment.

Reply:
Set owner reference here: https://github.com/hstreamdb/hstream-operator/blob/c1e4d162cd13df954be6ca0076d1251814c48261/internal/controller/connector_controller.go#L273.

Reply:
Great, but I still think we should add some labels, not just this deployment, for all of resource by controller created, just like in `internal/pod_helper.go`

Reply:
Got it. I know what you mean. Will supplement them soon.